src/arbitraries/types.ts

```diff
@@ -8,3 +8,8 @@ export type ArbitrarySize = {
   type: 'exact' | 'estimated'
   credibleInterval?: [number, number]
 }
+
+export type MappedArbitraryConfig<A,B> = {
+  inverseFunction?: (b: B) => A | A[],
```

Comment:
As mentioned above, return-type polymorphism is bad practice and in this case it won't deal correctly with arrays. You should either return always `A[]` or create different optional configs `injectiveInverseFunction` and `inverseFunction`.

Reply:
I know that it is not frequent (at least not to me) to have return-type polymorphism but I didn't know that it was a bad practice. Therefore I will change to return always `A[]` because it is a more simple and less verbose solution.

Reply:
Already added the modifications mentioned above.

Reply:
I already had discussed this with Hugo and we completely agree with Rui Gonçalves.